lang/rust/avro/src/schema_compatibility.rs

```diff
@@ -480,14 +421,7 @@ impl SchemaCompatibility {
                     );
                 }
                 SchemaKind::Duration => {
-                    if let Schema::Duration = writers_schema {
```

Comment:
Why this is not needed ?

Reply:
The same reason as above. When enter the pattern `SchemaKind::Duration`  is does not make sense the `if let` because we know before hand that both types are `Schema::Duration`, it can never raise an Error.